### PR TITLE
Add canonical package location to SwiftPMCacheKey to differentiate the same target when its package is changed from remote one to local one (or vice-versa)

### PR DESCRIPTION
**Pull request by ikesyo**
Currently there is no package information in SwiftPMCacheKey.

ref: https://github.com/swiftlang/swift-package-manager/blob/bc689418acad746af6f2e24e6bef640e411d9bd7/Sources/PackageModel/PackageIdentity.swift#L16-L19

### CODE REVIEW

**Review by freddi-kit — APPROVED** [review]

File: Sources/ScipioKit/Producer/Cache/CacheSystem.swift [hunk 1]
```diff
@@ -96,6 +96,7 @@ extension PinsStore.PinState: @retroactive Hashable {
 }
 
 public struct SwiftPMCacheKey: CacheKey {
+    public var packageIdentity: PackageIdentity
```
Comment (ikesyo) [inline Sources/ScipioKit/Producer/Cache/CacheSystem.swift:99]: ref: https://github.com/swiftlang/swift-package-manager/blob/bc689418acad746af6f2e24e6bef640e411d9bd7/Sources/PackageModel/PackageIdentity.swift#L16-L19

**Comment by ikesyo** [comment]
I fixed the implementation using [CanonicalPackageLocation](https://github.com/swiftlang/swift-package-manager/blob/bc689418acad746af6f2e24e6bef640e411d9bd7/Sources/PackageModel/PackageIdentity.swift#L345-L424) instead of PackageIdentity and added the test case.

File: Sources/ScipioKit/Producer/Cache/CacheSystem.swift [hunk 2]
```diff
@@ -96,6 +96,7 @@ extension PinsStore.PinState: @retroactive Hashable {
 }
 
 public struct SwiftPMCacheKey: CacheKey {
+    public var canonicalPackageLocation: String
```
Comment (giginet) [inline Sources/ScipioKit/Producer/Cache/CacheSystem.swift:99]: I'm worried about the remote URL being embedded. If it's used in a mirrored repository, all caches will be expired. I think it's better to be it `Optional` and it should have only for local packages.

Comment (ikesyo) [reply Sources/ScipioKit/Producer/Cache/CacheSystem.swift]: Thanks for the suggestion, I addressed this in 507be71.

**Review by giginet — APPROVED** [review]

**Review by freddi-kit — APPROVED** [review]

**Pull request merged**